src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -56,10 +58,24 @@ private GitHubUserCache(Docset docset, string token)
             var github = new GitHubAccessor(token);
             _getUserByLoginFromGitHub = github.GetUserByLogin;
             _getLoginByCommitFromGitHub = github.GetLoginByCommit;
-            _cachePath = string.IsNullOrEmpty(docset.Config.GitHub.UserCache)
-                ? AppData.GitHubUserCachePath
-                : docset.GetFileRestorePath(docset.Config.GitHub.UserCache);
             _expirationInHours = docset.Config.GitHub.UserCacheExpirationInHours;
+            var path = docset.Config.GitHub.UserCache;
+            if (string.IsNullOrEmpty(path))
+            {
+                _cachePath = AppData.DefaultGitHubUserCachePath;
+                return;
+            }
+            var restorePath = docset.GetFileRestorePath(path);
```

Comment:
>GetFileRestorePath [](start = 37, length = 18)

Suggestion: this method may return (restorePath, isFromURl) to avoid invoking `HrefUtility.IsHttpHref` twice.